homeassistant/components/playstation_network/config_flow.py

```diff
@@ -0,0 +1,116 @@
+"""Config flow for the Playstation Network integration."""
+
+import logging
+from typing import Any
+
+from psnawp_api.core.psnawp_exceptions import PSNAWPAuthenticationError
+from psnawp_api.psnawp import PSNAWP
+import voluptuous as vol
+
+from homeassistant import config_entries
+from homeassistant.config_entries import ConfigFlow, ConfigFlowResult
+from homeassistant.const import CONF_USERNAME
+from homeassistant.core import HomeAssistant
+from homeassistant.data_entry_flow import AbortFlow
+from homeassistant.exceptions import (
+    ConfigEntryAuthFailed,
+    ConfigEntryNotReady,
+    HomeAssistantError,
+)
+
+from .const import DOMAIN
+from .coordinator import PlaystationNetworkCoordinator
+
+_LOGGER = logging.getLogger(__name__)
+
+STEP_USER_DATA_SCHEMA = vol.Schema({vol.Required("pin"): str})
+
+
+async def validate_input(hass: HomeAssistant, data: dict[str, Any]) -> dict[str, Any]:
+    """Validate the user input allows us to connect.
+
+    Data has the keys from STEP_USER_DATA_SCHEMA with values provided by the user.
+    """
+    try:
+        npsso: str = str(data.get("pin"))
+        psn = PSNAWP(npsso)
+    except PSNAWPAuthenticationError as error:
+        raise ConfigEntryAuthFailed(error) from error
+    except Exception as ex:
+        raise ConfigEntryNotReady(ex) from ex
+
+    try:
+        user = await hass.async_add_executor_job(lambda: psn.user(online_id="me"))
+        client = psn.me()
+        coordinator = PlaystationNetworkCoordinator(hass, psn, user, client)
+        await coordinator.async_config_entry_first_refresh()
+    except PSNAWPAuthenticationError as error:
+        raise ConfigEntryAuthFailed(error) from error
+    except Exception as ex:
+        raise ConfigEntryNotReady(ex) from ex
+
+    # Return info that you want to store in the config entry.
+    return {
+        "title": "Playstation Network",
+        "npsso": npsso,
+        "username": coordinator.data.get("username"),
+        "data": data,
+    }
+
+
+class PlaystationNetworkConfigFlow(ConfigFlow, domain=DOMAIN):
+    """Handle a config flow for Playstation Network."""
+
+    VERSION = 1
+    CONNECTION_CLASS = config_entries.CONN_CLASS_CLOUD_POLL
```

Comment:
```suggestion
```

Version 1 is already the default, so not required.
What is `CONNECTION_CLASS` for? First time I see this in an integration and also it is deprecated since 2021.6.0 🤷